Add code style via ruff

m-reuter opened the pull request:
This switches on our code-style workflow mainly to run Ruff and spellchecker. So this fixes ruff messages in all code (lots of changes, but mainly very minor stuff).

m-reuter committed:
fix ruff warnings and formating

m-reuter committed:
sort include and other formatting fixes

m-reuter committed:
hypvinn fix formatting and ruff warnings

m-reuter committed:
include sort, fixes, formatting

m-reuter committed:
fastsurfercnn fix ruff warnings and cleanup

m-reuter committed:
test dir, fix ruff stuff

m-reuter committed:
run on PRs and pushes in dev

m-reuter committed:
correct spelling

m-reuter committed:
fix spelling

m-reuter committed:
add back (locally) unused imports in logging

m-reuter committed:
fix import order

m-reuter committed:
add back Logger and alias get_logger

m-reuter committed:
fix doc string bg_label description

m-reuter committed:
fix string split

m-reuter committed:
add back import of locally unused argparse in parser_defaults

m-reuter committed:
add back fields

dkuegler commented on Docker/install_fs_pruned.sh:
```diff
@@ -97,7 +97,7 @@ function run_parallel ()
 
 
-# get Freesurfer and upack (some of it)
+# get FreeSurfer and upnack (some of it)
```
```suggestion
# get FreeSurfer and unpack (some of it)
```

m-reuter committed:
fix typo

Co-authored-by: David Kügler <david.kuegler@dzne.de>

dkuegler committed:
Fix formatting in CerebNet/apply_warp.py

dkuegler committed:
HypVINN/run_prediction.py
- revert changes to the docstring of main
- move replacement of strings into a decorator function
- fix indentation errors in doc

FastSurferCNN/checkpoint.py
- import Scheduler from torch instead of string-declaring (which does not work with pipe for Union)

FastSurferCNN/parser_defaults.py
- revert <TYPE> | None to Optional[<TYPE>]
- add # noqa: UP0007 to ignore this ruff rule
- add documentation for this
- remove fields import
- add Optional import
- revert parser == None removal
- reformat for number of characters per line

m-reuter committed:
fix import sorting
